roofit/roofitcore/src/RooMinimizer.cxx

```diff
@@ -21,8 +21,7 @@
 RooMinimizer is a wrapper class around ROOT::Fit:Fitter that
 provides a seamless interface between the minimizer functionality
 and the native RooFit interface.
-By default the Minimizer is MINUIT for classic FcnMode and MINUIT2
-for gradient FcnMode.
+The Minimizer is MINUIT or MINUIT2 depending on configuration.
```

Comment:
> Thank you for the PR. I am not sure if the RooGradMinimizer class is needed in the case of the Autodifferentiation work, since it is a way to provide an external gradient to the Minimiser. Could you maybe split the removal of the RooGradMinimizer class in another PR, which we could merge it later when Jonas is back. The rest looks good to me.

Thanks for having considered this! For the AD mode, we have our own RooAbsMinimizerFcn implementation, so removing the RooGradMinimizerFcn is find to me, as I also said in https://github.com/root-project/root/pull/11716#pullrequestreview-1193963711.

@egpbos, maybe we can merge that other PR first, and then you can rebase this PR and address the review comments in one go?

Reply:
@guitargeek yes, that sounds good. I will start addressing the review comments here now. In the meantime, #11716 can be merged and once that is done I will rebase this branch on master.

@lmoneta Thanks for the correction. I have replaced the sentence with:
```
By default the Minimizer is MINUIT for classic mode and MINUIT2
for parallelized mode (activated with the `RooFit::NewStyle(true)`
parameter or by passing a RooRealL function as the minimization
target).
```